### ISSUE TITLE
Exporting files as SVG instead of PNG kinda works but not realy. any help?

### ISSUE THREAD

**Issue by chicky25** [HashLips/hashlips_art_engine#1048]
so i saw that canvas supports svg output. i added the following:

const canvas = createCanvas(format.width, format.height, '**svg**');
const ctx = canvas.getContext("2d");
var metadataList = [];
var attributesList = [];
var dnaList = [];

const buildSetup = () => {
  if (fs.existsSync(buildDir)) {
    fs.rmdirSync(buildDir, { recursive: true });
  }
  fs.mkdirSync(buildDir);
  fs.mkdirSync(path.join(buildDir, "/json"));
  fs.mkdirSync(path.join(buildDir, "/images"));
};

const getRarityWeight = (_str) => {
  let nameWithoutExtension = _str.slice(0, -4);
  var nameWithoutWeight = Number(
    nameWithoutExtension.split(rarityDelimiter).pop()
  );
  if (isNaN(nameWithoutWeight)) {
    nameWithoutWeight = 0;
  }
  return nameWithoutWeight;
};

const cleanDna = (_str) => {
  var dna = Number(_str.split(":").shift());
  return dna;
};

const cleanName = (_str) => {
  let nameWithoutExtension = _str.slice(0, -4);
  var nameWithoutWeight = nameWithoutExtension.split(rarityDelimiter).shift();
  return nameWithoutWeight;
};

const getElements = (path) => {
  return fs
    .readdirSync(path)
    .filter((item) => !/(^|\/)\.[^\/\.]/g.test(item))
    .map((i, index) => {
      return {
        id: index,
        name: cleanName(i),
        filename: i,
        path: `${path}${i}`,
        weight: getRarityWeight(i),
      };
    });
};

const layersSetup = (layersOrder) => {
  const layers = layersOrder.map((layerObj, index) => ({
    id: index,
    name: layerObj.name,
    elements: getElements(`${layersDir}/${layerObj.name}/`),
    blendMode:
      layerObj["blend"] != undefined ? layerObj["blend"] : "source-over",
    opacity: layerObj["opacity"] != undefined ? layerObj["opacity"] : 1,
  }));
  return layers;
};

const saveImage = (_editionCount) => {
  fs.writeFileSync(
    `${buildDir}/images/${_editionCount}.**svg**`,
    **canvas.toBuffer()**
  );
};

const genColor = () => {
  let hue = Math.floor(Math.random() * 360);
  let pastel = `hsl(${hue}, 100%, ${background.brightness})`;
  return pastel;
};

const drawBackground = () => {
  ctx.fillStyle = genColor();
  ctx.fillRect(0, 0, format.width, format.height);
};

const addMetadata = (_dna, _edition) => {
  let dateTime = Date.now();
  let tempMetadata = {
    dna: sha1(_dna.join("")),
    name: `PolyGerm #${_edition}`,
    description: description,
    image: `${baseUri}/${_edition}.**svg**`,
    edition: _edition,
    date: dateTime,
    ...extraMetadata,
    attributes: attributesList,
    compiler: "HashLips Art Engine",
  };
  metadataList.push(tempMetadata);
  attributesList = [];
};

and it looks like it creates the SVG files but they all have the same image in them
how can i fix this? or can i only use png output?

**Comment by bolshoytoster** [comment]
@chicky25
> but they all have the same image in them

what do you mean by this? Are they all the same?

**Comment by chicky25** [comment]
> @chicky25
> 
> > but they all have the same image in them
> 
> what do you mean by this? Are they all the same?

yes they look all the same

**Comment by bolshoytoster** [comment]
@chicky25 does it work with the same images in png format?

**Comment by chicky25** [comment]
> @chicky25 does it work with the same images in png format?

yes when i use the standard configuration it works flawlessly and generates all different pictures

**Comment by bolshoytoster** [comment]
@chicky25 I think you're basically on your own then.

If you could make a github repo of your project and link it here then I could try debugging it.

**Comment by chicky25** [comment]
> @chicky25 I think you're basically on your own then.
> 
> If you could make a github repo of your project and link it here then I could try debugging it.

ok thank you i will consider it
